### PR TITLE
fix(core): Improve community nodes loading

### PR DESCRIPTION
**Pull request by netroy**
1. use `require.resolve` to find the path for `n8n-nodes-base`
2. If a community package installs, but then fails to load, uninstall it.

### CODE REVIEW

**Comment by Joffcom** [comment]
Tested Adding, Deleting and Updating through the UI
Custom Extensions options with a manual install
Reinstall option with deleted nodes folder and deleted node_modules content

File: packages/cli/src/LoadNodesAndCredentials.ts [hunk 1]
```diff
@@ -110,31 +115,23 @@ export class LoadNodesAndCredentials implements INodesAndCredentials {
 	}
 
 	private async loadNodesFromBasePackages() {
-		const nodeModulesPath = await this.getNodeModulesPath();
-		const nodePackagePaths = await this.getN8nNodePackages(nodeModulesPath);
-
-		for (const packagePath of nodePackagePaths) {
-			await this.runDirectoryLoader(LazyPackageDirectoryLoader, packagePath);
-		}
+		await this.runDirectoryLoader(LazyPackageDirectoryLoader, NODES_BASE_DIR);
```
Comment (netroy) [inline packages/cli/src/LoadNodesAndCredentials.ts:118]: This change will break custom nodes installed in `node_modules`. But that's something I'd like us to do anyways, as letting users install their own packages in our `node_modules` can mess with n8n dependencies. Maybe this should wait until 1.0 release.

**Review by krynble — DISMISSED** [review]
Code looks good.

Tested starting n8n with missing package, it did alert me correctly.
Tested also initializing using `--reinstallMissingPackages` according to docs, worked fine.

Also tested initializing with missing package, n8n alerted. Clicked uninstall -> reinstall. Worked fine.

Also did install an outdated package, n8n warned about the updated, clicked update and it worked well.

Seems all right ✅

**Review by krynble — APPROVED** [review]
Ran some more tests, working also in queue mode as long as the community nodes were installed previously and accessible by worker.

Installing, removing, repairing, all working.

**Comment by janober** [comment]
Got released with [`n8n@0.222.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.222.0)

**Pull request merged**